future.go

```diff
@@ -113,3 +113,9 @@ func (f *SettableFuture) Catch(callback func(err error)) Future {
 	}
 	return f
 }
+
+// String converts the settable future to a string representation,
+// otherwise most testing frameworks will throw a data race
+func (f *SettableFuture) String() string {
+	return fmt.Sprintf("%p", f)
```

Comment:
That's familiar ;)  

I would just add the type name in there to make the stringer easier to understand if it shows up (likely accidentally) in program output.
```suggestion
	return fmt.Sprintf("SettableFuture<%p>", f)
```

Reply:
Very familiar 😆